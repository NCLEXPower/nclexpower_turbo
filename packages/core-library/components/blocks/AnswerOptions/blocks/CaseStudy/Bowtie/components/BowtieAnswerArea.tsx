import { Box } from "@mui/material";
import { useFormContext } from "react-hook-form";
import { ContainedCaseStudyQuestionType } from "../../../../../../../system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/types";
import { TextField } from "../../../../../../forms/TextField";

interface BowtieAnswerAreaProps {
  questionIndex: number;
}

export const BowtieAnswerArea = ({ questionIndex }: BowtieAnswerAreaProps) => {
  const { control, watch } = useFormContext<ContainedCaseStudyQuestionType>();

  return (
    <Box sx={{ display: "flex", flexDirection: "column", marginTop: "10px" }}>
      <Box sx={{ display: "flex", gap: 2 }}>
        <TextField
          control={control}
          name={`questionnaires.${questionIndex}.leftLabelName`}
          placeholder="Left Label"
        />
        <TextField
          control={control}
          name={`questionnaires.${questionIndex}.centerLabelName`}
          placeholder="Center Label"
        />
        <TextField
          control={control}
          name={`questionnaires.${questionIndex}.rightLabelName`}
          placeholder="Right Label"
        />
      </Box>
    </Box>
  );
};
