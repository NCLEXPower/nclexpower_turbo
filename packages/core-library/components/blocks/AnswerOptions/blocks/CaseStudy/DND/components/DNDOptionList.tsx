import { Box, Typography } from '@mui/material'
import React from 'react'
import { DndOptionsResponseType } from '../../../../../../../api/types'
import { IconButton } from '../../../../../../Button/IconButton'
import { EvaIcon } from '../../../../../../EvaIcon'

type Props = {
    optionList: DndOptionsResponseType[]
    deleteDndOption: (optionId: string) => void;
}

export const DNDOptionList: React.FC<Props> = ({ optionList, deleteDndOption }) => {
    return (
        <Box data-testid="dnd-option-list" gap={2} overflow="auto" height={1}>
            {optionList.length > 0 && optionList.map((option, index) =>
                <Box sx={{
                    boxShadow: '0px 3px 3px rgba(0, 0, 0, 0.3)',
                    textAlign: 'center',
                    backgroundColor: "#fff",
                    my: 2,
                    display: "flex",
                    justifyContent: "space-between",
                    alignItems: "center",
                    borderRadius: "10px",
                    p: 3
                }} key={option.id}>
                    <Typography variant='body1'>
                        {option.label}
                    </Typography>
                    <IconButton onClick={() => deleteDndOption(option.id)}>
                        <EvaIcon name='close-outline' width={13} height={13} />
                    </IconButton>
                </Box>
            )}
        </Box>
    )
}
