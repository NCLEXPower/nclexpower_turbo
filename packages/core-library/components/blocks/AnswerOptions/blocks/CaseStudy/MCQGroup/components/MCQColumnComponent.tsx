import { Box } from "@mui/material";
import { useFormContext } from "react-hook-form";
import { TextField } from "../../../../../../forms/TextField";

interface ColumnComponent {
  questionIndex: number;
  columnIndex: number;
}

export const ColumnComponent = ({
  columnIndex,
  questionIndex,
}: ColumnComponent) => {
  const { control } = useFormContext();
  return (
    <Box
      key={columnIndex}
      sx={{
        flexGrow: 1,
        border: 0.5,
        borderWidth: 0.5,
        minWidth: 150,
        maxWidth: 150,
      }}
    >
      <TextField
        name={`questionnaires.${questionIndex}.columns.${columnIndex}.label`}
        placeholder="Enter Column Name"
        control={control}
        sx={{
          width: "100%",
          "& fieldset": { border: "none" },
        }}
      />
    </Box>
  );
};
