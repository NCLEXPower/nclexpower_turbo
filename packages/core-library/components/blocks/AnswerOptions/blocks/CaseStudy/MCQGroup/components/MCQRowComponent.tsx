import { useFormContext } from "react-hook-form";
import { TablePropType } from "../../../../../../../system/app/internal/types";
import { ControlledCheckbox } from "../../../../../../Checkbox/Checkbox";
import { Box } from "@mui/material";
import { TextField } from "../../../../../../forms/TextField";

export const RowComponent: React.FC<TablePropType> = ({
  ColumnField,
  questionIndex,
  rowIndex,
}) => {
  const { control } = useFormContext();
  const ColumnFieldContainer = ColumnField.length > 0 && ColumnField.length - 1;

  return (
    <Box sx={{ display: "flex" }}>
      <Box
        sx={{
          flexGrow: 1,
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          border: 0.5,
          borderWidth: 0.5,
          minWidth: 150,
          maxWidth: 150,
        }}
      >
        <TextField
          control={control}
          name={`questionnaires.${questionIndex}.rows.${rowIndex}.rowTitle`}
          sx={{
            flexGrow: 1,
            display: "flex",
            minWidth: 150,
            maxWidth: 150,
            marginTop: -2,
          }}
          placeholder="Enter Text"
        />
      </Box>
      {Array(ColumnFieldContainer)
        .fill(false)
        .map((_, colIndex) => (
          <Box
            sx={{
              flexGrow: 1,
              display: "flex",
              justifyContent: "center",
              alignItems: "center",
              border: 0.5,
              borderWidth: 0.5,
              minWidth: 150,
              maxWidth: 150,
            }}
          >
            <ControlledCheckbox
              control={control}
              name={`questionnaires.${questionIndex}.rows.${rowIndex}.choices.${colIndex}.value`}
              defaultValue={0}
            />
          </Box>
        ))}
    </Box>
  );
};
