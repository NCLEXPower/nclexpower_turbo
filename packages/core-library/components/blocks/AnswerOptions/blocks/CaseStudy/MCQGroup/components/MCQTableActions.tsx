import React from "react";
import {
  MCQColumnType,
  MCQRowType,
} from "../../../../../../../system/app/internal/types";
import {
  initMCQColumn,
  initMCQRow,
} from "../../../../../../../system/app/internal/blocks/Hub/Settings/SettingsManagement/constants/constants";
import { useFieldArray, useFormContext } from "react-hook-form";
import { Button } from "../../../../../../Button/Button";

interface MCQTableConfigProps {
  RowField: MCQRowType[];
  ColumnField: MCQColumnType[];
  questionIndex: number;
}

export const MCQTableActions: React.FC<MCQTableConfigProps> = ({
  RowField,
  ColumnField,
  questionIndex,
}) => {
  const { setValue, control } = useFormContext();

  const { append: ColumnAppend, remove: ColumnRemove } = useFieldArray({
    control,
    name: `questionnaires.${questionIndex}.columns`,
  });

  const { append: RowAppend, remove: RowRemove } = useFieldArray({
    control,
    name: `questionnaires.${questionIndex}.rows`,
  });

  const handleRowAppendFields = () => {
    const nextRowId = RowField.length;
    const currentChoicesCount = ColumnField.length;
    const newChoices = Array.from(
      { length: currentChoicesCount - 1 },
      (_, index) => ({
        value: false,
        choiceId: index,
      })
    );

    const newRow = {
      ...initMCQRow,
      rowId: nextRowId,
      choices: newChoices,
    };

    RowAppend(newRow);
  };

  const handleColumnRemoveFields = (columnIndex: number | undefined) => {
    ColumnRemove(columnIndex);

    if (RowField.length > 0) {
      RowField.map((row: MCQRowType, index: number) => {
        const updatedChoices = Array.isArray(row.choices)
          ? row.choices.slice(0, row.choices.length - 1)
          : [];

        setValue(`questionnaires.${questionIndex}.rows.${index}`, {
          ...row,
          choices: updatedChoices,
        });
      });
    }
  };

  const handleColumnAppendFields = () => {
    ColumnAppend(initMCQColumn);

    if (RowField.length > 0) {
      RowField.map((row: MCQRowType, index: number) => {
        const updatedChoices = Array.isArray(row.choices)
          ? [
              ...row.choices,
              {
                value: false,
                choiceId: row.choices.length,
              },
            ]
          : [
              {
                value: false,
                choiceId: 0,
              },
            ];

        setValue(`questionnaires.${questionIndex}.rows.${index}`, {
          ...row,
          choices: updatedChoices,
        });
      });
    }
  };

  const ButtonFunctions = [
    {
      action: "add",
      count: ColumnField.length,
      setCount: () => handleColumnAppendFields(),
      limit: 2,
      label: "Add Column",
      type: "column",
    },
    {
      action: "add",
      count: RowField.length,
      setCount: () => handleRowAppendFields(),
      limit: 1,
      label: "Add Row",
      type: "row",
    },
    {
      action: "remove",
      count: RowField.length,
      setCount: () => RowRemove(RowField.length - 1),
      limit: 1,
      label: "Delete Row",
      type: "row",
    },
    {
      action: "remove",
      count: ColumnField.length,
      setCount: () => handleColumnRemoveFields(ColumnField.length - 1),
      limit: 3,
      label: "Delete Column",
      type: "column",
    },
  ];

  return ButtonFunctions.map((button) => (
    <Button
      data-testid="table-action-id"
      key={button.label}
      onClick={button.setCount}
      disabled={button.action === "remove" && button.count <= button.limit}
    >
      {button.label}
    </Button>
  ));
};
