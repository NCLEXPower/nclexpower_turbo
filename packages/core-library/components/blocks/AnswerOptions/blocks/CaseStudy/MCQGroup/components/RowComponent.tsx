import { useFormContext } from "react-hook-form";
import { TablePropType } from "../../../../../../../system/app/internal/types";
import { ControlledCheckbox } from "../../../../../../Checkbox/Checkbox";
import { useEffect } from "react";
import { Box } from "@mui/material";
import { TextField } from "../../../../../../forms/TextField";

export const RowComponent: React.FC<TablePropType> = ({
  column,
  questionIndex,
  rowIndex,
}) => {
  const { control, setValue, getValues } = useFormContext();

  useEffect(() => {
    const existingRowId = getValues(
      `questionnaires.${questionIndex}.rows.${rowIndex}.rowId`
    );
    if (!existingRowId) {
      setValue(
        `questionnaires.${questionIndex}.rows.${rowIndex}.rowId`,
        rowIndex && rowIndex + 1
      );
    }

    Array(column)
      .fill(null)
      .forEach((_, colIndex) => {
        const choicePath = `questionnaires.${questionIndex}.rows.${rowIndex}.choices.${colIndex}`;
        const existingChoiceValue = getValues(`${choicePath}.value`);

        if (existingChoiceValue === undefined) {
          setValue(`${choicePath}.choiceId`, colIndex);
          setValue(`${choicePath}.value`, false);
        }
      });
  }, [control, rowIndex, column, getValues, setValue]);

  return (
    <Box sx={{ display: "flex" }}>
      <Box
        sx={{
          flexGrow: 1,
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          border: 0.5,
          borderWidth: 0.5,
          minWidth: 150,
          maxWidth: 150,
        }}
      >
        <TextField
          control={control}
          name={`questionnaires.${questionIndex}.rows.${rowIndex}.rowTitle`}
          sx={{
            flexGrow: 1,
            display: "flex",
            minWidth: 150,
            maxWidth: 150,
            marginTop: -2,
          }}
          placeholder="Enter Text"
        />
      </Box>
      {Array(column)
        .fill(null)
        .map((_, colIndex) => (
          <Box
            sx={{
              flexGrow: 1,
              display: "flex",
              justifyContent: "center",
              alignItems: "center",
              border: 0.5,
              borderWidth: 0.5,
              minWidth: 150,
              maxWidth: 150,
            }}
          >
            <ControlledCheckbox
              control={control}
              name={`questionnaires.${questionIndex}.rows.${rowIndex}.choices.${colIndex}.value`}
              defaultValue={0}
            />
          </Box>
        ))}
    </Box>
  );
};
