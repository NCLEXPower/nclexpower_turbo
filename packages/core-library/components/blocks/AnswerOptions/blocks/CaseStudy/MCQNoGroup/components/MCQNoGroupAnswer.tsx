import React, { useState } from "react";
import {
  Box,
  FormControl,
  FormGroup,
  RadioGroup,
  Typography,
} from "@mui/material";
import { Control } from "react-hook-form";
import {
  TextField,
  EvaIcon,
  Button,
  ControlledRadio,
} from "../../../../../../";
import { ContainedCaseStudyQuestionType } from "../../../../../../../system/app/internal/blocks/Hub/Settings/SettingsManagement/steps/content/simulator/types";
import { ColumnComponent } from "../../MCQGroup/components/MCQColumnComponent";

type MCQNoGroupType = {
  questionIndex: number;
  handleAppendRowTable: () => void;
  disableAppendRow: boolean;
  handleAppendColumnHeaders: () => void;
  handleRemoveRow: (index: number) => void;
  handleRemoveColumnHeaders: (index: number) => void;
  columnHeaderFields: Array<{ label: string }>;
  control: Control<ContainedCaseStudyQuestionType>;
  tableRowFields: Array<{
    rowId?: number | undefined;
    rowTitle: string;
    choices: Array<{
      choiceId?: number;
      value: boolean;
    }>;
  }>;
  setValue: any;
  disableRemoveRow?: boolean;
  disableRemoveColumnHeaders?: boolean;
};

type SelectedValue = {
  rowIndex: number;
  colIndex: number;
};

export const MCQNoGroupAnswer: React.FC<MCQNoGroupType> = ({
  questionIndex,
  handleAppendRowTable,
  disableAppendRow,
  handleAppendColumnHeaders,
  handleRemoveRow,
  handleRemoveColumnHeaders,
  columnHeaderFields,
  tableRowFields,
  control,
  setValue,
  disableRemoveRow,
  disableRemoveColumnHeaders,
}) => {
  const handleRadioChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const checked = e.target.checked;
    const { colIndex, rowIndex } = JSON.parse(e.target.value) as SelectedValue;
    columnHeaderFields.slice(1).forEach((_, index) => {
      setValue(
        `questionnaires.${questionIndex}.rows.${rowIndex}.choices.${index}.value`,
        index === colIndex ? (checked ? true : false) : false
      );
    });
  };

  return (
    <Box>
      <Box
        sx={{
          width: "100%",
          display: "flex",
          justifyContent: "center",
          marginY: 2,
          gap: 1,
        }}
      >
        <Box
          display="grid"
          gridTemplateColumns="repeat(2, 1fr)"
          gap={2}
          width="100%"
        >
          <Button
            sx={{
              height: "45px",
              borderRadius: "10px",
              marginTop: "10px",
              width: "100%",
              textTransform: "none",
              gap: 1.5,
            }}
            onClick={handleAppendColumnHeaders}
            data-testid="append-column-headers"
          >
            <Typography variant="body2">Add Column</Typography>
            <EvaIcon
              name="plus-square-outline"
              fill="#fff"
              width={30}
              height={30}
              ariaHidden
            />
          </Button>
          <Button
            sx={{
              height: "45px",
              borderRadius: "10px",
              marginTop: "10px",
              width: "100%",
              textTransform: "none",
              gap: 1.5,
            }}
            onClick={handleAppendRowTable}
            data-testid="append-row-table"
            disabled={disableAppendRow}
          >
            <Typography variant="body2">Add Row</Typography>
            <EvaIcon
              name="plus-square-outline"
              fill="#fff"
              width={30}
              height={30}
              ariaHidden
            />
          </Button>
          <Button
            sx={{
              height: "45px",
              borderRadius: "10px",
              marginTop: "10px",
              width: "100%",
              textTransform: "none",
              gap: 1.5,
              backgroundColor: "#800f2f",
              "&:hover": {
                backgroundColor: "#800f2f95",
              },
            }}
            onClick={() =>
              handleRemoveColumnHeaders(columnHeaderFields.length - 1)
            }
            disabled={disableRemoveColumnHeaders}
            data-testid={`remove-column-header-${columnHeaderFields.length}`}
          >
            Remove Column
            <EvaIcon
              name="minus-square-outline"
              fill="#fff"
              width={30}
              height={30}
              ariaHidden
            />
          </Button>
          <Button
            sx={{
              height: "45px",
              borderRadius: "10px",
              marginTop: "10px",
              width: "100%",
              textTransform: "none",
              gap: 1.5,
              backgroundColor: "#800f2f",
              "&:hover": {
                backgroundColor: "#800f2f95",
              },
            }}
            onClick={() => handleRemoveRow(tableRowFields.length - 1)}
            data-testid={`remove-row-${tableRowFields.length}`}
            disabled={disableRemoveRow}
          >
            Remove Row
            <EvaIcon
              name="minus-square-outline"
              fill="#fff"
              width={30}
              height={30}
              ariaHidden
            />
          </Button>
        </Box>
      </Box>
      <Box
        sx={{
          overflowX: "auto",
          width: "100%",
          maxWidth: "550px",
          overflowY: "hidden",
          height: "fit-content",
        }}
      >
        <Box sx={{ display: "flex" }}>
          {columnHeaderFields?.length > 0 &&
            columnHeaderFields.map((_, index) => (
              <ColumnComponent
                key={`column-${index}`}
                columnIndex={index}
                questionIndex={questionIndex}
              />
            ))}
        </Box>

        <RadioGroup onChange={handleRadioChange}>
          {tableRowFields?.length > 0 &&
            tableRowFields.map((_, rowIndex) => (
              <Box key={`row-${rowIndex}`} sx={{ display: "flex" }}>
                <Box
                  sx={{
                    flexGrow: 1,
                    display: "flex",
                    justifyContent: "center",
                    alignItems: "center",
                    border: 0.5,
                    borderWidth: 0.5,
                    minWidth: 150,
                    maxWidth: 150,
                  }}
                >
                  <TextField
                    control={control}
                    name={`questionnaires.${questionIndex}.rows.${rowIndex}.rowTitle`}
                    sx={{
                      flexGrow: 1,
                      display: "flex",
                      minWidth: 150,
                      maxWidth: 150,
                      marginTop: -2,
                    }}
                    placeholder="Enter Text"
                  />
                </Box>
                {columnHeaderFields.slice(1).map((_, colIndex) => (
                  <Box
                    key={colIndex}
                    sx={{
                      flexGrow: 1,
                      display: "flex",
                      justifyContent: "center",
                      alignItems: "center",
                      border: 0.5,
                      borderWidth: 0.5,
                      minWidth: 150,
                      maxWidth: 150,
                    }}
                  >
                    <FormControl>
                      <FormGroup>
                        <ControlledRadio
                          name={`questionnaires.${questionIndex}.rows.${rowIndex}.choices.${colIndex}.value`}
                          value={JSON.stringify({ colIndex, rowIndex })}
                        />
                      </FormGroup>
                    </FormControl>
                  </Box>
                ))}
              </Box>
            ))}
        </RadioGroup>
      </Box>
    </Box>
  );
};
