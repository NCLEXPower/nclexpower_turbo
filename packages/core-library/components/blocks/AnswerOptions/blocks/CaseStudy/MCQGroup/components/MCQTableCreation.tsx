import { TablePropType } from "../../../../../../../system/app/internal/types";
import { Box } from "@mui/material";
import { RowComponent } from "./MCQRowComponent";
import { ColumnComponent } from "./MCQColumnComponent";

export const MCQTableCreation: React.FC<TablePropType> = ({
  ColumnField,
  RowField,
  questionIndex,
}) => {
  if (
    !ColumnField ||
    ColumnField.length <= 0 ||
    !RowField ||
    RowField.length <= 0
  ) {
    return "No data available";
  }

  return (
    <Box data-testid="mcq-table-id">
      <Box sx={{ display: "flex" }}>
        {ColumnField?.length > 0 &&
          ColumnField.map((_, index) => (
            <ColumnComponent
              columnIndex={index}
              questionIndex={questionIndex}
            />
          ))}
      </Box>

      {RowField?.length > 0 &&
        RowField.map((_, index) => (
          <RowComponent
            ColumnField={ColumnField}
            RowField={RowField}
            questionIndex={questionIndex}
            rowIndex={index}
          />
        ))}
    </Box>
  );
};
