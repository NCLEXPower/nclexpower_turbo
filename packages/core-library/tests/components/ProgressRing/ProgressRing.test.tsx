import { ProgressRing } from "core-library/components";
import "@testing-library/jest-dom";
import { render } from "../../common";

jest.mock("core-library/config", () => ({
  getConfig: jest
    .fn()
    .mockReturnValue({ publicRuntimeConfig: { processEnv: {} } }),
  config: { value: jest.fn() },
}));

jest.mock("core-library/core/router", () => ({
  useRouter: () => ({
    push: jest.fn(),
  }),
}));

describe("ProgressRing", () => {
  it("renders two CircularProgress components", () => {
    const { getAllByRole } = render(
      <ProgressRing
        value={75}
        size={60}
        color="#3f51b5"
        thickness={6}
        trackColor="#ccc"
      />
    );

    const progressBars = getAllByRole("progressbar");
    expect(progressBars).toHaveLength(2);
  });

  it("renders inner progress ring with correct value", () => {
    const { getAllByRole } = render(
      <ProgressRing value={80} size={50} color="green" />
    );

    const progressBars = getAllByRole("progressbar");

    expect(progressBars[1]).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <ProgressRing value={60} size={40} color="red" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
