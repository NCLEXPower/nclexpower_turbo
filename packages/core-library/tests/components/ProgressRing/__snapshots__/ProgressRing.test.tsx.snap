// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressRing matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-rmz4vk"
  >
    <span
      aria-valuenow="100"
      class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-xaqqdr-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px; transform: rotate(-90deg);"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="19.5"
          stroke-width="5"
          style="stroke-dasharray: 122.522; stroke-dashoffset: 0.000px;"
        />
      </svg>
    </span>
    <span
      aria-valuenow="60"
      class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-18ksbuk-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px; transform: rotate(-90deg);"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="19.5"
          stroke-width="5"
          style="stroke-dasharray: 122.522; stroke-dashoffset: 49.009px;"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
